pinot-common/src/main/java/org/apache/pinot/common/utils/JsonUtils.java

```diff
@@ -137,13 +138,15 @@ public static Object extractValue(@Nullable JsonNode jsonValue, FieldSpec fieldS
             }
             return values;
           } else {
-            return new Object[]{fieldSpec.getDefaultNullValue()};
+            return null;
```

Comment:
Don't know if we should support zero-length array for MV columns, but it is not the same as null.

Reply:
I dint follow. I am doing something similar to this in NullValueTransformer
```
          row.putField(fieldName, new Object[]{fieldSpec.getDefaultNullValue()});
```

Reply:
IMO empty array should be a valid value for MV columns instead of treated as null. We don't have that support now, but maybe better to add some comments or TODOs

Reply:
Want to maintain parity with the existing functionality. We can solve this in another PR.